packages/amplify_core/lib/src/types/exception/auth/signed_out_exception.dart

```diff
@@ -15,6 +15,14 @@ class SignedOutException extends AuthException {
     super.underlyingException,
   });
 
+  /// Thrown when no user is currently signed in.
+  const SignedOutException.noUserSignedIn({
```

Comment:
Should this be merged into the default constructor? I don't see why this case is special.